packages/compass-aggregations/src/components/stage-toolbar/option-menu.tsx

```diff
@@ -39,6 +39,51 @@ export const OptionMenu = ({
 }) => {
   const [menuOpen, setMenuOpen] = useState(false);
 
+  const menuItems = useMemo(() => {
+    return [
+      {
+        label: 'Add stage after',
+        onClick: () => {
+          onAddStageClick(index);
+          setMenuOpen(false);
+        },
+        icon: 'PlusWithCircle',
+      },
+      {
+        label: 'Add stage before',
+        onClick: () => {
+          onAddStageClick(index - 1);
+          setMenuOpen(false);
+        },
+        icon: 'PlusWithCircle',
+      },
+      {
+        label: 'Delete stage',
+        onClick: () => {
+          onDeleteStageClick(index);
+          setMenuOpen(false);
+        },
+        icon: 'Trash',
+      },
+      {
+        label: 'Expand documents',
+        onClick: () => {
+          onExpand(index);
+          setMenuOpen(false);
+        },
+        icon: 'ChevronDown',
+      },
+      {
+        label: 'Collapse documents',
+        onClick: () => {
+          onCollapse(index);
+          setMenuOpen(false);
+        },
+        icon: 'ChevronUp',
+      },
+    ];
+  }, [index, onAddStageClick, onDeleteStageClick, onExpand, onCollapse]);
+
```

Comment:
I've seen us doing this pattern elsewhere, but why do you see this being advantageous over simply rendering the children? I see possibilities of making this more DRY without falling back on this pattern of building component trees from arrays of "data".

Reply:
I added this to simply avoid adding `data-text` prop in the menu item because i did not want to repeat the text. 
How would you prefer to clean this?